File: AM/Lista1/w01java/app/src/main/java/com/example/w01_java/MainActivity.java
package com.example.w01_java;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;
import android.widget.Toast;

import java.util.Random;

public class MainActivity extends AppCompatActivity {

    private int r1 = 0;
    private int r2 = 0;
    private int score = 0;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        score = 0;
        roll();
    }

    private void roll() {
        ((TextView)findViewById(R.id.points)).setText("Punkty: " + score);
        Random r = new Random();
        r1 = r.nextInt(10);
        r2 = r.nextInt(10);
        ((Button)findViewById(R.id.leftClick)).setText("" + r1);
        ((Button)findViewById(R.id.rightClick)).setText("" + r2);
    }

    public void buttonLeft(View view) {
        if (r1 >= r2) {
            score++;
            Toast.makeText(this, "Dobrze", Toast.LENGTH_SHORT).show();
        } else {
            score--;
            Toast.makeText(this, "Zle", Toast.LENGTH_SHORT).show();
        }

        roll();
    }

    public void buttonRight(View view) {
        if (r2 >= r1) {
            score++;
            Toast.makeText(this, "Dobrze", Toast.LENGTH_SHORT).show();
        } else {
            score--;
            Toast.makeText(this, "Zle", Toast.LENGTH_SHORT).show();
        }

        roll();
    }
}

File: AM/Lista2/Hangman/app/src/main/java/com/example/hangman/MainActivity.kt
package com.example.hangman

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.ImageView
import android.widget.TextView
import android.widget.Toast
import androidx.constraintlayout.widget.ConstraintLayout
import androidx.core.content.ContextCompat
import androidx.core.view.children

class MainActivity : AppCompatActivity() {

    private val gameManager = GameManager()

    private lateinit var wordTextView: TextView
    private lateinit var lettersUsedTextView: TextView
    private lateinit var imageView: ImageView
    private lateinit var newGameButton: Button
    private lateinit var lettersLayout: ConstraintLayout

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        imageView = findViewById(R.id.imageView)
        wordTextView = findViewById(R.id.wordTextView)
        lettersUsedTextView = findViewById(R.id.lettersUsedTextView)
        newGameButton = findViewById(R.id.newGameButton)
        lettersLayout = findViewById(R.id.lettersLayout)

        newGameButton.setOnClickListener {
            startNewGame()
        }

        val gameState = gameManager.startNewGame(this)
        updateUI(gameState)

        lettersLayout.children.forEach { letterView ->
            if (letterView is TextView) {
                letterView.setOnClickListener {
                    val gameState = gameManager.play((letterView).text[0])
                    updateUI(gameState)
                }
            }
        }
    }

    private fun updateUI(gameState: GameState) {
        when (gameState) {
            is GameState.Lost -> gameOver(gameState.wordToGuess)
            is GameState.Running -> {
                wordTextView.text = gameState.hiddenWord
                lettersUsedTextView.text = "Letters used: ${gameState.lettersUsed}"
                imageView.setImageDrawable(ContextCompat.getDrawable(this, gameState.drawable))
            }
            is GameState.Won -> gameWon(gameState.wordToGuess)
        }
    }

    private fun gameOver(wordToGuess: String) {
        Toast.makeText(applicationContext, "Game Lost!", Toast.LENGTH_SHORT).show()
        wordTextView.text = wordToGuess
        imageView.setImageDrawable(ContextCompat.getDrawable(this, R.drawable.wisielec7))
    }

    private fun gameWon(wordToGuess: String) {
        Toast.makeText(applicationContext, "Game Won!", Toast.LENGTH_SHORT).show()
        wordTextView.text = wordToGuess
    }

    private fun startNewGame() {
        val gameState = gameManager.startNewGame(this)
        lettersLayout.visibility = View.VISIBLE
        lettersLayout.children.forEach { letterView ->
            letterView.visibility = View.VISIBLE
        }
        updateUI(gameState)
    }
}